dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -81,14 +81,18 @@
       {% if partitioned_by is not none %}
         {% set where_clauses = [] %}
         {% for column in partitioned_by %}
-          {% set partition_values = [] %}
           {% set values = run_query(get_partition_values(tmp_relation, column)) %}
+          {% set partition_values = [] %}
           {% for value in values %}
             {% do partition_values.append(column ~ "='" ~ value[0] ~ "'") %}
           {% endfor %}
           {% do where_clauses.append('(' ~ partition_values|join(' OR ') ~ ')') %}
         {% endfor %}
-        {% set partition_where_condition = where_clauses|join(' AND ') %}
+        {% if where_clauses | length > 0 %}
+            {% set partition_where_condition = where_clauses|join(' AND ') %}
+        {% else %}
+            {% set partition_where_condition = none %}
+        {% endif %}
```

Comment:
maybe in a perfect world we have the tenant where clause created up front _for this tenant_, then use that throughout, rather than trying to infer the where clause from the tmp table records.

Reply:
Yeah this is true, since it does a stage of all existing records and then deletes and re-inserts this would lead to it re-writing the whole table.

I can't think of a better approach than to just ditch the whole thing if there's nothing in the staging table, which feels hacky but will result in proper partition isolation. Updates incoming

Reply:
> maybe in a perfect world we have the tenant where clause created up front for this tenant, then use that throughout, rather than trying to infer the where clause from the tmp table records?

This would work for our models since our partition keys are always in variables, but that may not always be the case